### PR TITLE
Update NN-Vantage-Workflow.xml

### PR DESCRIPTION
**Pull request by nicoloereni**
* fix Workflow to avoid concurrency
Signed-off-by: Nicolò <nicolo.ereni@nozominetworks.com>

### CODE REVIEW

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml [hunk 1]
```diff
@@ -7,14 +17,14 @@
...
-<!--  In case you dont want to gev all the alerts, bookmark can be init to get only the alerts from the range passed  -->
-<!--  For exmple to get the Alerts from the last hour, bookmark as to be: -->
-<!--  <Initialize path="/bookmark" value="${time() - 3600000}" /> -->
-
   <Actions>
-    <Initialize path="/bookmark" value="0" />
-    <Initialize path="/bookmark_assets" value="0" />
-    <Initialize path="/size" value="100" />
+<!--    <Initialize path="/bookmark" value="0" />-->
+    <Initialize path="/bookmark" value="${time() - 7200000}" />
```
Comment (ChrisCollinsIBM) [inline Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:22]: Not required, but for clarity consider changing this to bookmark_alerts to help avoid confusion.

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml [hunk 2]
```diff
@@ -7,14 +17,14 @@
...
+<!--    <Initialize path="/bookmark_assets" value="0" />-->
+    <Initialize path="/bookmark_assets" value="${time() - 7200000}" />
+    <Initialize path="/size" value="1000" />
+    <Initialize path="/alert_page" value="1" />
+    <Initialize path="/asset_page" value="1" />
     <Initialize path="/first_run" value="true" />
```
Comment (ChrisCollinsIBM) [inline Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:28]: Not required, but for clarity consider changing this to first_run_alert to help avoid confusion.

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml [hunk 3]
```diff
@@ -47,37 +57,48 @@
...
       </If>
 
       <PostEvents path="/alerts/body/data" source="${/host}" />
 
       <Set path="/first_run" value="false" />
-      <If condition="max(/alerts/body/data/attributes/time) != null">
-        <Set path="/bookmark" value="${max(/alerts/body/data/attributes/time)}" />
+      <Set path="/alert_page" value="${/alert_page + 1}" />
+
+      <If condition="${count(/alerts/body/data)} != ${/size}">
```
Comment (ChrisCollinsIBM) [inline Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:84]: This method of determining if another page exists should be fairly reliable, but there would be a corner case where if there were exactly 1000 records available you might try and pull a second blank page.

That's probably ok but just wanted to mention it, if the API doesn't have anything else to inform you that there are additional pages then you'll probably have to stick with this method.

**Review by ChrisCollinsIBM — COMMENTED** [review]
Changes look good, I've left some feedback for further improvements but nothing that needs to be done to merge.

Thanks for the update!

**Review by ChrisCollinsIBM — APPROVED** [review]

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml (hunk 3 above)
Comment (nicoloereni) [reply Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:84]: Yes, it's exactly as you said. This is the easy/only way to do it right now.

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml (hunk 2 above)
Comment (nicoloereni) [reply Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:28]: yes, next version will have it

File: Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml (hunk 1 above)
Comment (nicoloereni) [reply Community Developed/NozomiNetworks/Vantage/NN-Vantage-Workflow.xml:22]: yes, next version will have it

**Pull request merged**